src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -34,372 +42,476 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
-                  call stop2(9992)
-               end if
-               call gsi_bundlegetpointer(en_bar(m),trim(cvars3d(ic3)),x3,istatus)
-               if(istatus/=0) then
-                  write(6,*)' error retrieving pointer to ',trim(cvars3d(ic3)),' for en_bar'
-                  call stop2(9993)
-               end if
-
-               select case (trim(cvars3d(ic3)))
-  
-                  case('sf','SF')
-     
-                     do k=1,grd_ens%nsig
-                        do i=1,grd_ens%lon2
-                           do j=1,grd_ens%lat2
-                              w3(j,i,k) = u(j,i,k)
-                              x3(j,i,k)=x3(j,i,k)+u(j,i,k)
-                           end do
-                        end do
-                     end do
-  
-                  case('vp','VP')
-  
-                     do k=1,grd_ens%nsig
-                        do i=1,grd_ens%lon2
-                           do j=1,grd_ens%lat2
-                              w3(j,i,k) = v(j,i,k)
-                              x3(j,i,k)=x3(j,i,k)+v(j,i,k)
-                           end do
-                        end do
-                     end do
-  
-                  case('t','T')
-  
-                     do k=1,grd_ens%nsig
-                        do i=1,grd_ens%lon2
-                           do j=1,grd_ens%lat2
-                              w3(j,i,k) = tv(j,i,k)
-                              x3(j,i,k)=x3(j,i,k)+tv(j,i,k)
-                           end do
-                        end do
-                     end do
-  
-                  case('q','Q')
-  
-                     do k=1,grd_ens%nsig
-                        do i=1,grd_ens%lon2
-                           do j=1,grd_ens%lat2
-                              w3(j,i,k) = rh(j,i,k)
-                              x3(j,i,k)=x3(j,i,k)+rh(j,i,k)
-                           end do
-                        end do
-                     end do
-  
-                  case('oz','OZ')
-  
-                     do k=1,grd_ens%nsig
-                        do i=1,grd_ens%lon2
-                           do j=1,grd_ens%lat2
-                              w3(j,i,k) = oz(j,i,k)
-                              x3(j,i,k)=x3(j,i,k)+oz(j,i,k)
-                           end do
-                        end do
-                     end do
-  
+ ! DEFINE INPUT FILE NAME
+           fv3_filename%grid_spec=trim(ensfilenam_str)//'-fv3_grid_spec' !exmaple thinktobe
+           fv3_filename%ak_bk=trim(ensfilenam_str)//'-fv3_akbk'
+           fv3_filename%dynvars=trim(ensfilenam_str)//'-fv3_dynvars'
+           fv3_filename%tracers=trim(ensfilenam_str)//"-fv3_tracer"
+           fv3_filename%sfcdata=trim(ensfilenam_str)//"-fv3_sfcdata"
+           fv3_filename%couplerres=trim(ensfilenam_str)//"-coupler.res"
+ ! 
+ ! READ ENEMBLE MEMBERS DATA
+           if (mype == 0) write(6,'(a,a)') 'CALL READ_FV3_REGIONAL_ENSPERTS FOR ENS DATA with the filename str : ',trim(ensfilenam_str)
```

Comment:
This line exceeds the 132 character limit.  Please reduce the number of characters on this line.